EmailNotificationActivity/__tests__/handler.test.ts

```diff
@@ -0,0 +1,146 @@
+      // tslint:disable-next-line: no-console
+      error: console.error,
+      // tslint:disable-next-line: no-console
+      info: console.log,
+      // tslint:disable-next-line: no-console
+      verbose: console.log,
+      // tslint:disable-next-line: no-console
+      warn: console.warn
+    }
+  } as any;
+
+  const aMessageId = "A_MESSAGE_ID" as NonEmptyString;
+
+  const aNewEmailNotification: NewNotification = {
+    channels: {
+      [NotificationChannelEnum.EMAIL]: {
+        addressSource: NotificationAddressSourceEnum.DEFAULT_ADDRESS,
+        toAddress: "to@example.com" as EmailAddress
+      }
+    },
+    fiscalCode: "FRLFRC74E04B157I" as any,
+    id: "A_NOTIFICATION_ID" as NonEmptyString,
+    kind: "INewNotification",
+    messageId: aMessageId
+  };
+
+  const aRetrievedNotification: RetrievedNotification = {
+    _etag: "_etag",
+    _rid: "_rid",
+    _self: "_self",
+    _ts: 1,
+    ...aNewEmailNotification,
+    kind: "IRetrievedNotification"
+  };
+
+  const notificationModelMock = ({
+    find: jest.fn(() => taskEither.of(some(aRetrievedNotification)))
+  } as unknown) as NotificationModel;
+
+  const aNotificationId = "A_NOTIFICATION_ID" as NonEmptyString;
+  const anOrganizationFiscalCode = "00000000000" as OrganizationFiscalCode;
+
+  const aMessageBodyMarkdown = "test".repeat(80) as MessageBodyMarkdown;
+
+  const aMessageContent: MessageContent = {
+    markdown: aMessageBodyMarkdown,
+    subject: "test".repeat(10) as MessageSubject
+  };
+
+  const aMessage = {
+    createdAt: new Date(),
+    fiscalCode: "FRLFRC74E04B157I" as any,
+    id: aMessageId,
+    indexedId: aMessageId,
+    kind: "INewMessageWithoutContent" as "INewMessageWithoutContent",
+    senderServiceId: "s123" as NonEmptyString,
+    senderUserId: "u123" as NonEmptyString,
+    timeToLiveSeconds: 3600 as TimeToLiveSeconds
+  };
+
+  const aSenderMetadata: CreatedMessageEventSenderMetadata = {
+    departmentName: "IT" as NonEmptyString,
+    organizationFiscalCode: anOrganizationFiscalCode,
+    organizationName: "AgID" as NonEmptyString,
+    requireSecureChannels: false,
+    serviceName: "Test" as NonEmptyString,
+    serviceUserEmail: "email@example.com" as EmailAddress
+  };
+
+  const HTML_TO_TEXT_OPTIONS: HtmlToTextOptions = {
+    ignoreImage: true, // ignore all document images
+    tables: true
+  };
+
+  const MAIL_FROM = "IO - l’app dei servizi pubblici <no-reply@io.italia.it>" as NonEmptyString;
+  const defaultNotificationParams = {
+    HTML_TO_TEXT_OPTIONS,
+    MAIL_FROM
+  };
+
+  const input: EmailNotificationActivityInput = {
+    notificationEvent: {
+      content: aMessageContent,
+      message: aMessage,
+      notificationId: aNotificationId,
+      senderMetadata: aSenderMetadata
+    }
+  };
+
+  it("should respond with 'SUCCESS' if the mail is sent", async () => {
+    const lMailerTransporterMock = ({
+      sendMail: jest.fn((_, f) => f(null, taskEither.of("SUCCESS")))
+    } as unknown) as MailerTransporter;
```

Comment:
By mocking the transport's `sendMail` you are _cheating_, as you are using the `mailer`'s inner implementation (something you assume to work). Try mocking `mailer`'s `sendMail` method instead. 
At this level, we only need to test `mailer`'s `sendMail` boundaries (that is: all its possible results).

Reply:
but you need a mock (`lMailerTransporterMock`) for getEmailNotificationActivityHandler:

   ```
const emailNotificationActivityHandler = getEmailNotificationActivityHandler(
      lMailerTransporterMock,
      notificationModelMock,
      defaultNotificationParams
    );

```

Reply:
you can mock it as well, just passing anything that comply with its type definition. However, you are not supposed to play with it's implementation to test `mailer.sendMail`